resource_customizations/apps/Deployment/actions/scale/action.lua

```diff
@@ -0,0 +1,4 @@
+local os = require("os")
```

Comment:
In ArgoCD, there are automatic tests that run for each built-in resource action.
Those tests expect an input manifest, and and output manifest, that represents the output of the action when run on the input manifest.
Look at `resource_customizations/apps/Deployment/actions/action_test.yaml` , it has an example of input and output files for the action test, you would need to add a similar for Deployment + StatefulSet. The idea is that you provide a Deployment manifest with replicas set to x as an input, and provide a Deployment manifest with replicas set to y as an output. The automatic test will succeed if your custom action yields the output manifest.
You don't have to do this twice for scale-in and scale-out, one example of a scale change is enough.

Reply:
I have added scale actions in `resource_customizations/apps/Deployment/actions/action_test.yaml` and `resource_customizations/apps/StatefulSet/actions/action_test.yaml` files and also tests in `util/lua/custom_actions_test.go`

Reply:
Hi @reggie-k, could you please take a look into this? it's green I believe.